packages/common/src/store/purchase-content/sagas.ts

```diff
@@ -911,12 +931,30 @@ function* purchaseWithAnything({
       throw new Error('Failed to fetch USDC user bank token account info')
     }
 
-    // Get the solana wallet provider
-    const provider = window.solana
-    if (!provider) {
-      throw new Error('No solana provider / wallet found')
+    let sourceWallet: PublicKey
+
+    const isNativeMobile = yield* getContext('isNativeMobile')
+    const mobileWalletActions = yield* getContext('mobileWalletActions')
+    if (isNativeMobile && mobileWalletActions) {
+      const { connect } = mobileWalletActions
+      const getWalletConnectPublicKey = (state: any) =>
+        state.walletConnect.publicKey
+      const dappKeyPair = nacl.box.keyPair()
```

Comment:
should create this and persist it via redux-persist, so you only have to "connect" once

Reply:
this can be a follow up